crates/task-impls/src/network.rs

```diff
@@ -286,156 +286,262 @@ impl<
         membership: &TYPES::Membership,
     ) {
         let mut maybe_action = None;
-        let (sender, message_kind, transmit): (_, _, TransmitType<TYPES>) =
-            match event.as_ref().clone() {
-                HotShotEvent::QuorumProposalSend(proposal, sender) => {
-                    maybe_action = Some(HotShotAction::Propose);
-                    (
-                        sender,
-                        MessageKind::<TYPES>::from_consensus_message(SequencingMessage::General(
-                            GeneralConsensusMessage::Proposal(proposal),
-                        )),
-                        TransmitType::Broadcast,
-                    )
-                }
+        let Some((sender, message_kind, transmit)) =
+            self.parse_event(event, &mut maybe_action, membership).await
+        else {
+            return;
+        };
+        self.spawn_transmit_task(message_kind, membership, maybe_action, transmit, sender);
```

Comment:
nit to get rid of else
```suggestion
        if let Some((sender, message_kind, transmit)) =
            self.parse_event(event, &mut maybe_action, membership).await {
            self.spawn_transmit_task(message_kind, membership, maybe_action, transmit, sender);
        };
```